File: practicorpc/05-grpc/03-framework-python-rpc/rpyc/b-calculadora/calculator_client.py
import rpyc
import sys


def _read(prompt: str) -> str:
    if sys.stdin.isatty():
        sys.stdout.write(prompt)
        sys.stdout.flush()
    return sys.stdin.readline().strip()


def main() -> None:
    opcion = _read("Operación (suma, resta, divide, multiplica): ").lower()
    op1 = int(_read("Operando 1: "))
    op2 = int(_read("Operando 2: "))

    connection = rpyc.connect("localhost", 18862)
    try:
        match opcion:
            case "suma":
                result = connection.root.suma(op1, op2)
            case "resta":
                result = connection.root.resta(op1, op2)
            case "divide":
                result = connection.root.divide(op1, op2)
            case "multiplica":
                result = connection.root.multiplica(op1, op2)
            case _:
                raise ValueError("Operación inválida")

        print("Resultado:", result)
    finally:
        connection.close()


if __name__ == "__main__":
    main()

File: practicorpc/04-java-rmi/b-paridad-y-calculadora/calculadora/src/ar/edu/ssdd/rmi/calculadora/CalculatorClient.java
package ar.edu.ssdd.rmi.calculadora;

import java.rmi.Naming;
import java.util.Scanner;

public class CalculatorClient {
    public static void main(String[] args) throws Exception {
        String host = (args.length > 0) ? args[0] : "localhost";
        CalculatorService service = (CalculatorService) Naming.lookup("rmi://" + host + "/CalculatorService");

        try (Scanner scanner = new Scanner(System.in)) {
            boolean tty = System.console() != null;
            if (tty) System.out.print("Operación (1=suma, 2=resta, 3=divide, 4=multiplica): ");
            int option = scanner.nextInt();

            if (tty) System.out.print("Operando 1: ");
            int op1 = scanner.nextInt();

            if (tty) System.out.print("Operando 2: ");
            int op2 = scanner.nextInt();

            int result = switch (option) {
                case 1 -> service.suma(op1, op2);
                case 2 -> service.resta(op1, op2);
                case 3 -> service.divide(op1, op2);
                case 4 -> service.multiplica(op1, op2);
                default -> throw new IllegalArgumentException("Opción inválida");
            };

            System.out.println("Resultado: " + result);
        }
    }
}
